analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -184,4 +188,14 @@ private ProgramState ProcessMethod(SymbolicContext context, IMethodSymbol method
         }
         return state;
     }
+
+    private static ProgramState ProcessIndexerAccess(ProgramState state, IPropertyReferenceOperationWrapper propertyReference)
+    {
+        state = state.SetOperationConstraint(propertyReference.Instance, CollectionConstraint.NotEmpty);
+        if (state.ResolveCaptureAndUnwrapConversion(propertyReference.Instance).TrackedSymbol() is { } symbol)
```

Comment:
I'd use the ternary operator here. Same readability, less useless () {}s

Reply:
I personally find the if a tiny bit easier to understand. But it definitely adds a bunch of clutter.